onadata/apps/main/views.py

```diff
@@ -1102,7 +1103,10 @@ def form_photos(request, username, id_string):
 
             for i in ['small', 'medium', 'large', 'original']:
                 url = reverse(attachment_url, kwargs={'size': i})
-                url = '%s?media_file=%s' % (url, attachment.media_file.name)
+                url = '%s?media_file=%s' % (
+                    url,
+                    quote_plus(attachment.media_file.name),
+                )
```

Comment:
Attachment model has a `secure_url()`. What about using it? 
We could do something like:
```
        data = {}
        data['original'] = attachment.secure_url()
        for suffix in settings.THUMB_CONF.keys():
            data[suffix] = attachment.secure_url(suffix)
```

Reply:
Great. Tested on `kc.master`